api/src/main/java/jakarta/enterprise/inject/build/compatible/spi/ScannedClasses.java

```diff
@@ -0,0 +1,25 @@
+package jakarta.enterprise.inject.build.compatible.spi;
+
+/**
+ * Allows registering additional classes to be scanned during bean discovery.
+ * Annotations on these classes can later be transformed using {@link Enhancement @Enhancement}.
+ */
+public interface ScannedClasses {
+    /**
+     * Adds a class with given name to the set of classes to be scanned during bean discovery.
+     *
+     * @param className binary name of the class, as defined by <cite>The Java&trade; Language Specification</cite>;
+     * in other words, the class name as returned by {@link Class#getName()}
+     */
+    void add(String className);
+
+    /**
+     * Adds all subclasses of the class with given name to the set of classes to be scanned during bean discovery.
+     *
+     * @param className binary name of the class, as defined by <cite>The Java&trade; Language Specification</cite>;
+     * in other words, the class name as returned by {@link Class#getName()}
+     */
+    // TODO adds the type itself or not? (in theory yes, as subtyping is reflexive)
+    // TODO looks like it can't be implemented on top of Portable Extensions, so maybe remove?
+    void addSubtypesOf(String className);
```

Comment:
Can we remove this method? It seems to violate that only annotated types should be added

Reply:
Yea we really should. I find that useful in certain scenarios, but it can't be implemented on top of Portable Extensions either, so it should go. I'll amend the PR.

Reply:
Done.